File: sample/src/main/java/tech/thdev/androidrecyclerview/adapter/model/AdapterContract.java
package tech.thdev.androidrecyclerview.adapter.model;

import tech.thdev.support.widget.adapter.model.BaseRecyclerModel;
import tech.thdev.support.widget.data.BaseItem;

/**
 * Created by Tae-hwan on 10/10/2016.
 */

public interface AdapterContract {

    interface View {

        void reload();
    }

    interface Model<T extends BaseItem> extends BaseRecyclerModel<T> {

    }
}

File: sample/src/main/java/tech/thdev/androidrecyclerview/ui/header_footer/adapter/model/AdapterContract.kt
package tech.thdev.androidrecyclerview.ui.header_footer.adapter.model

import tech.thdev.support.widget.adapter.model.BaseRecyclerModel
import tech.thdev.support.widget.data.BaseItem

/**
 * Created by Tae-hwan on 10/10/2016.
 */
interface AdapterContract {
    interface View {
        fun reload()
    }

    interface Model<T : BaseItem> : BaseRecyclerModel<T>
}